modules/wallet/encrypt_test.go

```diff
@@ -293,6 +302,45 @@ func TestInitFromSeedParallelUnlock(t *testing.T) {
 	}
 }
 
+// TestUnlockConcurrent verifies that calling unlock multiple times
+// concurrently results in only one unlock operation.
+func TestUnlockConcurrent(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+	// create a wallet with some money
+	wt, err := createWalletTester(t.Name())
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer wt.closeWt()
+
+	// lock the wallet
+	wt.wallet.Lock()
+
+	// spawn an unlock goroutine
+	done := make(chan struct{})
+	go func() {
+		// acquire the write lock so that Unlock acquires the trymutex, but
+		// cannot proceed further
+		wt.wallet.mu.Lock()
+		wt.wallet.Unlock(wt.walletMasterKey)
```

Comment:
shouldn't this error be getting checked? IIRC calling t.Fatal in a goroutine is a race condition so maybe the best would be to change `done` to a `chan error`, pass this error down the channel and check it in the main thread